### ISSUE TITLE
Peek fill up all my swap

### ISSUE THREAD

**Issue by cedlemo** [phw/peek#2]
After stopping the recording, my system have a hudge load and my swap if filled up.
My system is ArchLinux x86_64 , and I have 6 G ram and 2G swap.

**Comment by phw** [comment]
Might be related to the fact that the screen recording uses temp files, and by default /tmp on Arch is a RAM disk. What approximate duration and area was your recording?

I will look into this and see whether I can reduce the used storage.

**Comment by mareknovotny** [comment]
how could default /tmp be changed to something else?

**Comment by phw** [comment]
Try to start Peek like this to use the `tmp` directory in your home (or set any other path you want): 

    TMPDIR=$HOME/tmp peek

**Comment by mareknovotny** [comment]
thanks! it works fine

**Comment by phw** [comment]
And if you don't want to have /tmp use your RAM at all but use the disk instead you need to disable tmpfs, see https://wiki.archlinux.org/index.php/Tmpfs for pointers on how to do this.

**Comment by lolilolicon** [comment]
@phw Is there a reason why you're using huffyuv output? It's ridiculously HUGE.

I know nothing about the matter, but see [the comment on this page](https://panthema.net/2014/0630-Frame-Perfect-Linux-Screencasts-in-the-Year-2014.html) recommending libx264rgb, for instance.

Personally, I use PNG output (e.g. `%05d.png`) and post-process with ImageMagick or Gifsicle.

**Comment by phw** [comment]
> @phw Is there a reason why you're using huffyuv output? It's ridiculously HUGE.

No, no particular reason other than that it worked. I had also a short try with using PNG output, but was not too satisfied with the output I got. But actually I did not invest any more time in trying to optimize this after I had first satisfactory quality results :) Thanks for the link, will definitely change this.

**Comment by phw** [comment]
Ok, finally found the time to test this and use libx264rgb instead of huffyuv output. huffyuv is indeed ridiculously large. For comparison I recorded a roughly 900 x 500 pixel area, and I recorded about 500 frames. With huffyuv the temporary file was about 300 MB large, with libx264rgb only 3 MB. I think this should solve most RAM issues encountered, and everything else not fixed by this can still use the `TMPDIR=$HOME/tmp peek` workaround.
